src/dc_mimefactory.rs

```diff
@@ -1094,8 +1096,17 @@ unsafe fn get_subject(
 ) -> *mut libc::c_char {
     let context = (*chat).context;
     let ret: *mut libc::c_char;
-    let raw_subject =
-        dc_msg_get_summarytext_by_raw((*msg).type_0, (*msg).text, &mut (*msg).param, 32, context);
+
+    let raw_subject = {
+        let msgtext_c = (*msg)
+            .text
+            .as_ref()
+            .map(|s| CString::yolo(String::as_str(s)));
+        let msgtext_ptr = msgtext_c.map_or(ptr::null(), |s| s.as_ptr());
```

Comment:
Oh, this is nice!  I think there are a lot of places that simply do `s.as_ptr()` instead of the full `.map_or(ptr::null(), |s| s.as_ptr())` and that's probably the cause of a bunch of crashes.